GC on Azure fails for repos with storage namespace of the form https://{stAccount}.blob.core.windows.net/{container}

GC on Azure fails for repositories with storage namespace of the form https://{stAccount}.blob.core.windows.net/{container} with the following exception:

```
org.apache.spark.SparkException: [FAILED_EXECUTE_UDF] Failed to execute user defined function (GarbageCollector$$$Lambda$4409/1015546575: (string) => array<struct<_1:string,_2:binary,_3:binary>>)
	at org.apache.spark.sql.errors.QueryExecutionErrors$.failedExecuteUserDefinedFunctionError(QueryExecutionErrors.scala:220)
	at org.apache.spark.sql.errors.QueryExecutionErrors.failedExecuteUserDefinedFunctionError(QueryExecutionErrors.scala)
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage2.generate_doConsume_0$(Unknown Source)
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage2.hashAgg_doAggregateWithKeysOutput_0$(Unknown Source)
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage2.hashAgg_doAggregateWithKeys_0$(Unknown Source)
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage2.processNext(Unknown Source)
	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:759)
	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
	at org.apache.spark.shuffle.sort.BypassMergeSortShuffleWriter.write(BypassMergeSortShuffleWriter.java:140)
	at org.apache.spark.shuffle.ShuffleWriteProcessor.write(ShuffleWriteProcessor.scala:59)
	at org.apache.spark.scheduler.ShuffleMapTask.$anonfun$runTask$3(ShuffleMapTask.scala:81)
	at com.databricks.spark.util.ExecutorFrameProfiler$.record(ExecutorFrameProfiler.scala:110)
	at org.apache.spark.scheduler.ShuffleMapTask.$anonfun$runTask$1(ShuffleMapTask.scala:81)
	at com.databricks.spark.util.ExecutorFrameProfiler$.record(ExecutorFrameProfiler.scala:110)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:53)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:41)
	at org.apache.spark.scheduler.Task.doRunTask(Task.scala:168)
	at org.apache.spark.scheduler.Task.$anonfun$run$1(Task.scala:136)
	at com.databricks.spark.util.ExecutorFrameProfiler$.record(ExecutorFrameProfiler.scala:110)
	at org.apache.spark.scheduler.Task.run(Task.scala:96)
	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$13(Executor.scala:870)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1690)
	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:873)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at com.databricks.spark.util.ExecutorFrameProfiler$.record(ExecutorFrameProfiler.scala:110)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:728)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: shadegooglecommon.util.concurrent.UncheckedExecutionException: scala.MatchError: [Ljava.lang.String;@9f39938 (of class [Ljava.lang.String;)
	at shadegooglecommon.cache.LocalCache$Segment.get(LocalCache.java:2051)
	at shadegooglecommon.cache.LocalCache.get(LocalCache.java:3953)
	at shadegooglecommon.cache.LocalCache.getOrLoad(LocalCache.java:3976)
	at shadegooglecommon.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:4960)
	at io.treeverse.clients.ApiClient.getStorageNamespace(ApiClient.scala:150)
	at io.treeverse.clients.ApiClient.getMetaRangeURL(ApiClient.scala:183)
	at io.treeverse.clients.GarbageCollector$.getRangeTuples(GarbageCollector.scala:81)
	at io.treeverse.clients.GarbageCollector$.$anonfun$getRangesDFFromCommits$1(GarbageCollector.scala:101)
	... 28 more
Caused by: scala.MatchError: [Ljava.lang.String;@9f39938 (of class [Ljava.lang.String;)
	at io.treeverse.clients.ApiClient$.translateURI(ApiClient.scala:74)
	at io.treeverse.clients.ApiClient.keyToStorageNamespace(ApiClient.scala:140)
	at io.treeverse.clients.ApiClient$$anon$2.load(ApiClient.scala:131)
	at io.treeverse.clients.ApiClient$$anon$2.load(ApiClient.scala:130)
	at shadegooglecommon.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3529)
	at shadegooglecommon.cache.LocalCache$Segment.loadSync(LocalCache.java:2278)
	at shadegooglecommon.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2155)
	at shadegooglecommon.cache.LocalCache$Segment.get(LocalCache.java:2045)
	... 35 more
```

This is happening due to a parsing error in https://github.com/treeverse/lakeFS/blob/b241f78441f0d722861860db09add0fddbdae769/clients/spark/core/src/main/scala/io/treeverse/clients/ApiClient.scala#L74

### Steps to reproduce
1. Use a lakefs setup with azure blob as the underlying storage
2. create a repo with storage namespace of the form https://{stAccount}.blob.core.windows.net/{container} (i.e, without blob after the container part)
3. Configure retention rules for the repo
4. run gc for the repo

--- a/clients/spark/core/src/main/scala/io/treeverse/clients/ApiClient.scala
+++ b/clients/spark/core/src/main/scala/io/treeverse/clients/ApiClient.scala
@@ -71,7 +71,11 @@ private object ApiClient {
        *  extract the storage account, container and blob path, and use them in abfs url
        */
       val storageAccountName = StorageUtils.AzureBlob.uriToStorageAccountName(uri)
-      val Array(_, container, blobPath) = uri.getPath.split("/", 3)
+      val (container, blobPath) = uri.getPath.split("/", 3) match {
+        case Array(a, b, c) => (b, c)
+        case Array(a, b)    => (b, "")
+        case _              => throw new IllegalArgumentException
+      }
       return new URI(
         s"abfs://${container}@${storageAccountName}.dfs.core.windows.net/${blobPath}"
       )

--- a/clients/spark/core/src/main/scala/io/treeverse/clients/ApiClient.scala
+++ b/clients/spark/core/src/main/scala/io/treeverse/clients/ApiClient.scala
@@ -74,7 +74,10 @@ private object ApiClient {
       val (container: String, blobPath: String) = uri.getPath.split("/", 3) match {
         case Array(a, b, c) => (b, c)
         case Array(a, b)    => (b, "")
-        case _              => throw new IllegalArgumentException(s"Expected https://host/container or https://host/container/path in ${uri.toString}")
+        case _ =>
+          throw new IllegalArgumentException(
+            s"Expected https://host/container or https://host/container/path in ${uri.toString}"
+          )
       }
       return new URI(
         s"abfs://${container}@${storageAccountName}.dfs.core.windows.net/${blobPath}"
@@ -145,7 +148,7 @@ class ApiClient private (conf: APIConfigurations) {
           .normalize()
           .toString
       case StorageClientType.SDKClient => repo.getStorageNamespace
-      case _                           => throw new IllegalArgumentException("Unknown storage type ${key.storageClientType}")
+      case _ => throw new IllegalArgumentException("Unknown storage type ${key.storageClientType}")
     }
     storageNamespace
   }

--- a/clients/spark/core/src/main/scala/io/treeverse/clients/ApiClient.scala
+++ b/clients/spark/core/src/main/scala/io/treeverse/clients/ApiClient.scala
@@ -71,10 +71,10 @@ private object ApiClient {
        *  extract the storage account, container and blob path, and use them in abfs url
        */
       val storageAccountName = StorageUtils.AzureBlob.uriToStorageAccountName(uri)
-      val (container, blobPath) = uri.getPath.split("/", 3) match {
+      val (container: String, blobPath: String) = uri.getPath.split("/", 3) match {
         case Array(a, b, c) => (b, c)
         case Array(a, b)    => (b, "")
-        case _              => throw new IllegalArgumentException
+        case _              => throw new IllegalArgumentException(s"Expected https://host/container or https://host/container/path in ${uri.toString}")
       }
       return new URI(
         s"abfs://${container}@${storageAccountName}.dfs.core.windows.net/${blobPath}"
@@ -145,7 +145,7 @@ class ApiClient private (conf: APIConfigurations) {
           .normalize()
           .toString
       case StorageClientType.SDKClient => repo.getStorageNamespace
-      case _                           => throw new IllegalArgumentException
+      case _                           => throw new IllegalArgumentException("Unknown storage type ${key.storageClientType}")
     }
     storageNamespace
   }

